app/services/users/delete.rb

```diff
@@ -10,7 +10,7 @@ def call
       delete_user_activity
       user.unsubscribe_from_newsletters
       CacheBuster.bust("/#{user.username}")
-      user.delete
+      user.destroy
```

Comment:
The intention here was to delete all the associations manually and avoid triggering a lot of callbacks while `destroy`ing the user which could make the destroying take much longer (and it would be nice to avoid relying on the callbacks magic here overall).
Is there value here in destroying instead of deleting?

Reply:
Given we have destroyed many of the associations already this should no longer take a long time. The advantage here is that we rely on A LOT on callbacks in the models so to destroy it here would ensure any of those small callbacks we may have forgotten about get triggered. Until we have eliminated all callbacks I think using destroy is a much safer and more consistent option that will eliminate bugs and surprises

Reply:
> Until we have eliminated all callbacks

![](https://media.giphy.com/media/mpZOT04g1ETm0/source.gif)